rust/src/write.rs

```diff
@@ -277,6 +293,20 @@ impl<'a, W: Write + Seek> Writer<'a, W> {
             .entry(header.channel_id)
             .or_insert(0) += 1;
 
+        // if the current chunk is larger than our target chunk size, finish it
+        // and start a new one.
+        let current_chunk_size = match &self.writer {
+            Some(WriteMode::Chunk(cw)) => Some(cw.compressor.position()),
+            _ => None,
+        };
+        if let (Some(current_chunk_size), Some(target)) =
+            (current_chunk_size, self.options.chunk_size)
+        {
+            if current_chunk_size > target {
+                self.finish_chunk()?;
+            }
+        }
+
         self.chunkin_time()?.write_message(header, data)?;
```

Comment:
> self.chunkin_time

wut

Reply:
It's ~~turtles~~ [memes](https://knowyourmeme.com/memes/its-morbin-time) all the way down.